### PR TITLE
gradle-jenkins-plugin

### PR DESCRIPTION
**Pull request by renatoavt**

### CODE REVIEW

File: _posts/2016/2016-06-14-gradle-jenkins-plugin.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+date: "2016-06-14 10:59:11"
+image: blog-header/gradle-jenkins-tools.jpg
+categories: tech-stories
+authors: ["Renato"]
+---
+
+Since the [last blog post]( https://developer.epages.com/blog/2016/01/28/jenkins-job-dsl-plugin.html) post we did some updates in our Jenkins environments and today I'm going to talk a bit about it.
+We are using a [Gradle Jenkins plugin](https://github.com/ghale/gradle-jenkins-plugin) to control the Jenkins installation.
+
+We already mention about the DSL on the previous blogpost, you can find more information about the [Job DSL Plugin](https://developer.epages.com/blog/2016/01/28/jenkins-job-dsl-plugin.html)
```
Comment (RyanScovilleEpages) [inline _posts/2016/2016-06-14-gradle-jenkins-plugin.md]: We already mention about the
->
We already mentioned the

File: _posts/2016/2016-06-14-gradle-jenkins-plugin.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+So you may ask why we changed it?
```
Comment (RyanScovilleEpages) [inline _posts/2016/2016-06-14-gradle-jenkins-plugin.md]: So you may ask why we changed it?
->
"Why did we change it?" you may ask.

File: _posts/2016/2016-06-14-gradle-jenkins-plugin.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+For the other plugin you need to create a seed job, and at ePages, we have quite complex structure with a few shell scripts.
```
Comment (RyanScovilleEpages) [inline _posts/2016/2016-06-14-gradle-jenkins-plugin.md]: we have quite complex structure
->
we have quite a complex structure

File: _posts/2016/2016-06-14-gradle-jenkins-plugin.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+Some of those shell scripts are shared on the company level, so before changing them, we want to make sure they work properly. (testing phase!)
```
Comment (RyanScovilleEpages) [inline _posts/2016/2016-06-14-gradle-jenkins-plugin.md]: (testing phase!)
->
(Testing phase!)

File: _posts/2016/2016-06-14-gradle-jenkins-plugin.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+Another advantage of this plugin is, that it is easy to deploy. For instance I didn't want to break our Team Jenkins installation, so I run a instance of Jenkins in a docker container and deploy my playground scripts to it.
```
Comment (RyanScovilleEpages) [inline _posts/2016/2016-06-14-gradle-jenkins-plugin.md]: this plugin is, that
->
this plugin is that

File: _posts/2016/2016-06-14-gradle-jenkins-plugin.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+    } // end dsl
+}
+{% endhighlight %}
+
+*<font color='red'>Note:</font>* This will not setup a Jenkins for you! It will only authenticate with your own instance!
+
+## How to use the plugin
+
+I was using on my local Jenkins. After doing the proper changes on the DSL, I just need to trigger the task updateJenkinsItems.
```
Comment (RyanScovilleEpages) [inline _posts/2016/2016-06-14-gradle-jenkins-plugin.md]: I was using on my local Jenkins.
->
I was using my local Jenkins.

File: _posts/2016/2016-06-14-gradle-jenkins-plugin.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+./gradlew updateJenkinsItems
+{% endhighlight %}
+This will update your Jenkins installation with the DSL defined locally.
+
+-----------------------
+{% image blog/blog-gradlew_updateJenkinsItems.jpg %}
+-----------------------
+
+## Conclusions
+This plugin is really powerful, and there are many cases where you could use it to empower your jobs/automation environment. I hope the blog post illustrates the application examples of this plugin.
```
Comment (RyanScovilleEpages) [inline _posts/2016/2016-06-14-gradle-jenkins-plugin.md]: I hope the blog
->
I hope this blog

File: _posts/2016/2016-06-14-gradle-jenkins-plugin.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+If you want to know more about the other jobs, have a look at complete documentation on the [project wiki](https://github.com/ghale/gradle-jenkins-plugin/wiki).
+
+### Advantages
+- We have our Jenkins jobs under a source control system (git).
+- We can recreate our Jenkins instance pretty fast and test locally without affecting the rest of the team.
```
Comment (RyanScovilleEpages) [inline _posts/2016/2016-06-14-gradle-jenkins-plugin.md]: without affecting
->
without effecting

File: _posts/2016/2016-06-14-gradle-jenkins-plugin.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+### Disadvantages
+- If you are not familiar with Gradle, it might take some time for you to become acquainted with the configuration.
+
+
+### Credits
+We didn't come up with a company-wide solution, but use the plugin in our team and adapted it to our needs. So a big thanks to [Gary Hale](https://github.com/ghale), because without him I wouldn't be writing about it today{% emoji wink %}.
```
Comment (RyanScovilleEpages) [inline _posts/2016/2016-06-14-gradle-jenkins-plugin.md]: and adapted it
->
and adapt it

**Comment by renatoavt** [comment]
https://github.com/ePages-de/epages-docs/pull/547

**Pull request closed without merging**